eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,121 @@
+# Description: This script is used to verify the specs used to generate SDK package point to the main branch of Azure/azure-rest-api-specs repository
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $ServiceDirectory,
+  [Parameter(Position = 1)]
+  [ValidateNotNullOrEmpty()]
+  [string] $PackageName,
+  [Parameter(Position = 2)]
+  [ValidateNotNullOrEmpty()]
+  [string]$GitHubPat
+)
+
+. $PSScriptRoot/common.ps1
+. $PSScriptRoot/Helpers/PSModule-Helpers.ps1
+. $PSScriptRoot/Invoke-GitHubAPI.ps1
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+
+# This function is used to verify the 'require' and 'input-file' settings in autorest.md point to the main branch of Azure/azure-rest-api-specs repository
+function Verify-AutorestMd([string]$fileUrl) {
+  if($fileUrl -match "^https://github.com/(?<repo>[^/]*/azure-rest-api-specs)/blob/(?<commit>[0-9a-f]{40})/(?<path>.*)") {
+    $repo = $matches['repo']
+    $commit = $matches['commit']
+    if($repo -ne "Azure/azure-rest-api-specs") {
+      LogError "Invalid repo in the file url: $fileUrl. Repo should be 'Azure/azure-rest-api-specs'."
+      exit 1
+    }
+    # check the commit hash belongs to main branch
+    Verify-CommitFromMainBranch $commit
+  }
+  else{
+    LogError "Invalid file url: $fileUrl"
+    exit 1
+  }
+}
+
+# This function is used to verify the 'repo' and 'commit' settings in tsp-location.yaml point to the main branch of Azure/azure-rest-api-specs repository
+function Verify-TspLocation([System.Object]$tspLocationObj) {
+  $repo = $tspLocationObj["repo"]
+  $commit = $tspLocationObj["commit"]
+  if($repo -ne "Azure/azure-rest-api-specs") {
+    LogError "Invalid repo setting in the tsp-location.yaml: $repo. Repo should be 'Azure/azure-rest-api-specs'."
+    exit 1
+  }
+
+  # check the commit hash belongs to main branch
+  Verify-CommitFromMainBranch $commit
+}
+
+# This function is used to verify the specific 'commit' belongs to the main branch of Azure/azure-rest-api-specs repository
+function Verify-CommitFromMainBranch([string]$commit) {
+  try {
+    $searchResult = Search-GitHubCommit -AuthToken $GitHubPat -CommitHash "0f39a2d56070d2bc4251494525cb8af88583a938" -RepoOwner "Azure" -RepoName "azure-rest-api-specs"
```

Comment:
Is the PAT needed here? Are we worried about rate limiting?

Reply:
I prefer to provide the PAT given the rate limit is three times difference. 10 calls vs 30 calls.  I would like to avoid failing the pipeline by this rate limit issue.